### DISCUSSION TITLE
External reference to codes

### DISCUSSION THREAD

**Discussion by SergejMuhic** [jwouellette/TPF-5_372-Unit_Test_Suite#12] [Ideas]
As discussed in the meeting on the 10th January 2024 we discussed two options but actually have 3 options:
1. Using [Project Document Information](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Project_Context/Project_Document_Information/content.html) which attaches `IfcDocumentInformation` to `IfcProject` and [Document Association](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Object_Association/Document_Association/content.html) which attaches the actual document to as I understand the requirement of the AASHTO Pset to `IfcProject`
2. Keep the references in a Pset by using explicit properties (e. g. RailCode, BridgeCode, WhateverCode) of type [`IfcPropertyReferenceValue`](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcPropertyReferenceValue.htm).
3. Have an external reference directly on the `IfcPropertySingleValue` (actually on its root `IfcPropertyAbstraction`.`HasExternalReferences`) in question. So if the property is `CodeReference` it could have a list of external references to the codes.

Options 2 and 3 are not clean and are also much more cumbersome to implement. I think option 1 is the way to go.

**Comment by jwouellette** [comment]
IF using option #1, should we also move all document references to the IfcProject or IfcBridge level?

**Comment by SergejMuhic** [comment]
Maybe the bridge codes on IfcBridge, road on IfcRoad, rail on IfcRail etc.?

**Reply by RickBrice** [reply]
In a project with multiple bridges, each different type of bridge (cable stay, segmental, steel, timber, etc) could have different design specifications and requirements. Attaching requirements directly to the bridge to which they apply seems like a good idea.

**Reply by jwouellette** [reply]
@RickBrice should we make the Construction Documentation (PDF) a reference on the IfcBridge level as well, rather than on the IfcProject level? It seems that if a "project" may contain multiple bridges/assets, and each asset has its own CD set, then it should be attached at the IfcFacilty/IfcBridge level.

**Reply by RickBrice** [reply]
That seems logical but I'm no expert on this.
